### PR TITLE
refactor: the 'base' and a few 'header' components

### PR DESCRIPTION
**Pull request by orlinmalkja**

### CODE REVIEW

File: src/components/base/BaseIcon.vue [hunk 1]
```diff
@@ -1,7 +1,12 @@
-<script setup>
+<script setup lang="ts">
 import { getIcon } from '@/utils/icons';
 
-const props = defineProps(['name']);
+//const props = defineProps(['name']);
+export interface Props {
+  name: string
```
Comment (orlinmalkja) [inline src/components/base/BaseIcon.vue:6]: Should I set the type of the name to be part of a list of names, or just let it string ?

File: src/components/header/GlobalHeader.vue [hunk 2]
```diff
@@ -16,27 +16,28 @@
...
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  configErrorTitle: ''
+})
 
-const show = computed(() => config.value?.header?.show);
-const manifests = computed(() => store.getters['contents/manifests']);
+const store = useStore();
 const config = computed(() => store.getters['config/config']);
```
Comment (orlinmalkja) [inline src/components/header/GlobalHeader.vue:37]: Is it necessary to create an interface and assign it as a type to 'config' ?

File: src/components/base/BaseIcon.vue (hunk 1 above)
Comment (paulpestov) [reply src/components/base/BaseIcon.vue:6]: string should be enough

File: src/components/base/BaseIcon.vue [hunk 3]
```diff
@@ -1,7 +1,12 @@
-<script setup>
+<script setup lang="ts">
 import { getIcon } from '@/utils/icons';
 
-const props = defineProps(['name']);
+//const props = defineProps(['name']);
```
Comment (paulpestov) [inline src/components/base/BaseIcon.vue:4]: unused comment

**Review by paulpestov — APPROVED** [review]

**Pull request merged**